src/blockedtuple.jl

```diff
@@ -0,0 +1,76 @@
+# This file defines BlockedTuple, a Tuple of heterogeneous Tuple with a BlockArrays.jl
+# like interface
+
+using BlockArrays: Block, BlockArrays, BlockIndexRange, BlockRange, blockedrange
+
+struct BlockedTuple{Divs,Flat}
+  flat::Flat
```

Comment:
```suggestion
struct BlockedTuple{BlockLengths,Flat} <: AbstractBlockTuple
  flat::Flat
```
to match the BlockArrays terminology.

Also I think we should make an `AbstractBlockTuple` supertype and define as much code as possible on `AbstractBlockTuple` so that we can define subtypes, for example `AbstractBlockedPermutation`/`BlockedPermutation` could be `AbstractBlockTuple` subtypes which would help with sharing code.